Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/GenericCommandExecutionAndEventPublication.php

```diff
@@ -57,66 +85,182 @@ abstract protected function readPayloadTable(TableNode $payloadTable): array;
+        }
+        if (is_string($commandArguments['sourceOriginDimensionSpacePoint'] ?? null) && !empty($commandArguments['sourceOriginDimensionSpacePoint'])) {
+            $commandArguments['sourceOriginDimensionSpacePoint'] = OriginDimensionSpacePoint::fromJsonString($commandArguments['sourceOriginDimensionSpacePoint'])->coordinates;
+        } elseif (!isset($commandArguments['sourceOriginDimensionSpacePoint'])) {
+            $commandArguments['sourceOriginDimensionSpacePoint'] = $this->currentDimensionSpacePoint?->coordinates;
+        }
+        if (isset($commandArguments['succeedingSiblingNodeAggregateId']) && $commandArguments['succeedingSiblingNodeAggregateId'] === '') {
+            unset($commandArguments['succeedingSiblingNodeAggregateId']);
+        }
+        if (is_string($commandArguments['nodeAggregateId'] ?? null) && str_starts_with($commandArguments['nodeAggregateId'], '$')) {
+            $commandArguments['nodeAggregateId'] = $this->rememberedNodeAggregateIds[substr($commandArguments['nodeAggregateId'], 1)]?->value;
+        } elseif (!isset($commandArguments['nodeAggregateId'])) {
+            $commandArguments['nodeAggregateId'] = $this->currentNodeAggregate?->nodeAggregateId->value;
+        }
+        if (is_string($commandArguments['sourceNodeAggregateId'] ?? null) && str_starts_with($commandArguments['sourceNodeAggregateId'], '$')) {
+            $commandArguments['sourceNodeAggregateId'] = $this->rememberedNodeAggregateIds[substr($commandArguments['sourceNodeAggregateId'], 1)]?->value;
+        } elseif (!isset($commandArguments['sourceNodeAggregateId'])) {
+            $commandArguments['sourceNodeAggregateId'] = $this->currentNodeAggregate?->nodeAggregateId->value;
+        }
+        if (is_string($commandArguments['parentNodeAggregateId'] ?? null) && str_starts_with($commandArguments['parentNodeAggregateId'], '$')) {
+            $commandArguments['parentNodeAggregateId'] = $this->rememberedNodeAggregateIds[substr($commandArguments['parentNodeAggregateId'], 1)]?->value;
+        } elseif (!isset($commandArguments['parentNodeAggregateId'])) {
+            $commandArguments['parentNodeAggregateId'] = $this->currentNodeAggregate?->nodeAggregateId->value;
+        }
+        if (is_string($commandArguments['tetheredDescendantNodeAggregateIds'] ?? null)) {
+            if ($commandArguments['tetheredDescendantNodeAggregateIds'] === '') {
+                unset($commandArguments['tetheredDescendantNodeAggregateIds']);
+            } else {
+                $commandArguments['tetheredDescendantNodeAggregateIds'] = json_decode($commandArguments['tetheredDescendantNodeAggregateIds'], true, 512, JSON_THROW_ON_ERROR);
+            }
+        }
+        if (is_array($commandArguments['references'] ?? null)) {
+            $commandArguments['references'] = iterator_to_array($this->mapRawNodeReferencesToNodeReferencesToWrite($commandArguments['references']));
+        }
```

Comment:
this is definitely the weak spot of this PR. I'll clean it up if tests are green

Reply:
done with 50742546a20ace297dbe7f9425e182d9266291e6